src/frontend/js/hooks/useLearnerCoursesSearch/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Enrollment, CredentialOrder, OrderState, ProductType } from 'types/Joanie';
+import { Maybe, Nullable } from 'types/utils';
+import { useOrdersEnrollments } from 'pages/DashboardCourses/useOrdersEnrollments';
+
+const useLearnerCoursesSearch = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [count, setCount] = useState<Maybe<number>>(0);
+  const [orderAndEnrollmentList, setOrderAndEnrollmentList] = useState<
+    (CredentialOrder | Enrollment)[]
+  >([]);
+  const [isNewSearchLoading, setIsNewSearchLoading] = useState(false);
+  const query = searchParams.get('query') || undefined;
+  const {
+    data,
+    isLoading,
+    next,
+    hasMore,
+    count: currentCount,
+    error,
+  } = useOrdersEnrollments({
+    query,
+    orderFilters: {
+      product_type: [ProductType.CREDENTIAL],
+      state_exclude: [OrderState.CANCELED],
+    },
+  });
+
+  useEffect(() => {
+    if (!data.length && isLoading) {
+      setIsNewSearchLoading(true);
+    }
+
+    if (isLoading) {
+      return;
+    }
+
+    if (isNewSearchLoading) {
+      setIsNewSearchLoading(false);
+    }
+
+    if (isNewSearchLoading || data.length > orderAndEnrollmentList?.length) {
+      setOrderAndEnrollmentList(data as (CredentialOrder | Enrollment)[]);
+
+      // research counter should not be displayed when query is empty
+      if (query) {
+        setCount(currentCount);
+      }
+    }
+  }, [data.length, isLoading, isNewSearchLoading, query]);
+
+  const submitSearch = (newQuery: Nullable<string>) => {
+    if (newQuery === null) {
+      searchParams.delete('query');
+    } else {
+      searchParams.set('query', newQuery);
+    }
+
+    setSearchParams(searchParams);
+    if (!newQuery) {
+      setCount(undefined);
+    }
+  };
+
+  return {
+    submitSearch,
+    data: orderAndEnrollmentList,
```

Comment:
why not using `data` directly?

Reply:
`data` is the return or `useOrdersEnrollments` where `orderAndEnrollmentList` is an insternal state.
`orderAndEnrollmentList` isn't up to date each time data is, it's not a bug but a feature:
When `query` chagne,  `useOrdersEnrollments` queries are invalidated and `data` is temporary set to `0`.
There we don't want the list to disepear but to fade until data get a new value. That the usage of the internal state.